go-controller/pkg/ovn/ipallocator/allocator/bitmap.go

```diff
@@ -234,3 +247,27 @@ func (contiguousScanStrategy) AllocateBit(allocated *big.Int, max, count int) (i
 }
 
 var _ bitAllocator = contiguousScanStrategy{}
+
+// roundRobinScanStrategy tries to allocate starting at 0 and scans forward looking for
+// the next available address, wrapping the range if necessary. It caches the
+// last allocation and will try to allocate all other addresses before re-allocating
+// the most recently allocated address.
+type roundRobinScanStrategy struct {
+	lastAllocated int
+}
+
+func (wss *roundRobinScanStrategy) AllocateBit(allocated *big.Int, max, count int) (int, bool) {
+	if count >= max {
+		return 0, false
+	}
+	for i := 0; i < max; i++ {
```

Comment:
```suggestion
	for i := 1; i < max; i++ {
```

per https://github.com/ovn-org/ovn-kubernetes/pull/1502/files#r455516372

Reply:
I would rather keep that outer loop the same, but change:

`at := (wss.lastAllocated + 1  + i) % max`.

Reply:
@abhat @aojea done slightly differently (see fixup) because we the way suggested here leaves the 0 bit in the map unallocated until it wraps around, unlike the other methods. Changed "last allocated" to "next to allocate" since the core problem was lastAllocated could not distinguish between the never-allocated-anything and only-allocated-0-bit cases.

PTAL

Reply:
makes sense.